common/network/lock_handle.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __COMMON_NETWORK_LOCK_HANDLE_H__
```

Comment:
these locks etc. are not useful just for networking stuff.
let's put this in `common/lock_handle.h`?

Reply:
(same for cond_handle)